docs/sources/helm-charts/mimir-distributed/configure/configure-autoscaling.md

```diff
@@ -0,0 +1,194 @@
+{{< /admonition >}}
+
+This topic explains how to configure autoscaling for Mimir components using KEDA and Kubernetes Horizontal Pod Autoscaler (HPA).
+
+## Before you begin
+
+- Ensure you have a running Mimir cluster deployed with Helm
+- Verify you have the required permissions to modify Helm deployments
+- Familiarize yourself with Kubernetes HPA concepts
+
+## Prerequisites
+
+To use autoscaling, you need:
+
+1. KEDA installed in your Kubernetes cluster
+2. Prometheus metrics available for scaling decisions
+
+{{< admonition type="warning" >}}
+Do not use the same Mimir or Grafana Enterprise Metrics cluster for storing and querying autoscaling metrics. Using the same cluster can create a dangerous feedback loop:
+
+1. If the Mimir/GEM cluster becomes unavailable, autoscaling will stop working because it cannot query the metrics
+2. This can prevent the cluster from automatically scaling up during high load or recovery
+3. The inability to scale can further exacerbate the cluster's unavailability
+4. This may prevent the cluster from recovering
+
+Instead, use a separate Prometheus instance or a different metrics backend for autoscaling metrics.
+{{< /admonition >}}
+
+## Supported components
+
+Mimir Helm chart supports autoscaling for the following components:
+
+- [Querier](https://grafana.com/docs/mimir/<MIMIR_VERSION>/references/architecture/components/querier/)
+- [Query-frontend](https://grafana.com/docs/mimir/<MIMIR_VERSION>/references/architecture/components/query-frontend/)
+- [Distributor](https://grafana.com/docs/mimir/<MIMIR_VERSION>/references/architecture/components/distributor/)
+
+## About KEDA
+
+KEDA is a Kubernetes operator that simplifies the setup of HPA with custom metrics from Prometheus. It consists of:
+
+- An operator and external metrics server
+- Support for multiple metric sources, including Prometheus
+- Custom resources (`ScaledObject`) that define scaling parameters
+- Automatic HPA resource management
+
+For more information, refer to [KEDA documentation](https://keda.sh).
+
+## Configure autoscaling for a new installation
+
+This section describes how to enable autoscaling when deploying Mimir for the first time.
+
+### Steps
+
+1. Configure the Prometheus metrics source in your values file:
+
+   ```yaml
+   kedaAutoscaling:
+     prometheusAddress: "http://prometheus.monitoring:9090"
+     pollingInterval: 10
+   ```
+
+2. Enable and configure autoscaling for desired components:
+
+   ```yaml
+   querier:
+     kedaAutoscaling:
+       enabled: true
+       minReplicaCount: 2
+       maxReplicaCount: 10
+   ```
+
+3. Deploy Mimir using Helm:
+   ```bash
+   helm upgrade --install mimir grafana/mimir-distributed -f values.yaml
+   ```
+
+### Expected outcome
+
+After deployment:
+
+- KEDA creates ScaledObject resources for configured components
```

Comment:
`ScaledObject` ?

Reply:
that's the name of the resource 🤷 https://keda.sh/docs/2.14/concepts/scaling-deployments/#scaledobject-spec

similar to Service, Pod, Deployment and StatefulSet

Reply:
```suggestion
- KEDA creates `ScaledObject` resources for configured components
```